From f3f3991a9756b8bde84561ed1365c74f28b1ef8d Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 12 May 2018 11:35:22 -0500
Subject: [PATCH 8/8] Finish the missing code to handle VS in test_common.in

.. and a *load* of other test files.

Clearly this is not the best way, need to instead change
execdir to include VS and be done with it.

Now the testsuite should be able to run.
---
 CMakeLists.txt                     |  1 +
 configure.ac                       |  1 +
 dap4_test/test_data.sh             |  4 ++--
 dap4_test/test_meta.sh             |  4 ++--
 dap4_test/test_parse.sh            |  2 +-
 dap4_test/test_remote.sh           |  2 +-
 dap4_test/tst_data.sh              |  4 ++--
 dap4_test/tst_meta.sh              |  4 ++--
 dap4_test/tst_parse.sh             |  2 +-
 examples/C/run_examples.sh         | 12 ++++++------
 examples/C/run_examples4.sh        | 10 +++++-----
 examples/C/run_filter.sh           |  2 +-
 examples/CDL/do_comps.sh           |  8 +++++---
 hdf4_test/run_formatx_hdf4.sh      |  2 +-
 hdf4_test/run_get_hdf4_files.sh    |  2 +-
 nc_test/run_diskless.sh            | 14 +++++++-------
 nc_test/run_diskless5.sh           |  2 +-
 nc_test/run_mmap.sh                |  8 ++++----
 nc_test4/filter_test/tst_filter.sh |  4 ++--
 nc_test4/run_bm.sh                 |  6 +++---
 nc_test4/run_bm_ar4.sh             | 10 +++++-----
 nc_test4/run_bm_elena.sh           | 12 ++++++------
 nc_test4/run_bm_test1.sh           | 14 +++++++-------
 nc_test4/run_bm_test2.sh           |  8 ++++----
 nc_test4/run_empty_vlen_test.sh    |  2 +-
 nc_test4/run_grp_rename.sh         |  6 +++---
 nc_test4/run_tst_chunks.sh         |  4 ++--
 nc_test4/tst_filter.sh             |  4 ++--
 nc_test4/tst_szip.sh               |  6 +++---
 ncdap_test/tst_remote.sh           |  2 +-
 ncdump/run_utf8_tests.sh           |  2 +-
 ncdump/tst_bom.sh                  |  4 ++--
 ncdump/tst_dimsizes.sh             |  2 +-
 ncdump/tst_fileinfo.sh             |  2 +-
 ncdump/tst_fillbug.sh              |  2 +-
 ncdump/tst_h_scalar.sh             |  2 +-
 ncdump/tst_lengths.sh              | 14 +++++++-------
 ncdump/tst_nccopy4.sh              |  4 ++--
 ncdump/tst_netcdf4.sh              | 22 +++++++++++-----------
 ncdump/tst_netcdf4_4.sh            |  2 +-
 ncdump/tst_output.sh               |  4 ++--
 test_common.in                     |  4 ++--
 42 files changed, 115 insertions(+), 111 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93e2806a..14903695 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1876,6 +1876,7 @@ FILE(COPY ${netCDF_SOURCE_DIR}/include/netcdf_meta.h
 SET(EXTRA_DIST ${EXTRA_DIST} ${CMAKE_SOURCE_DIR}/test_common.in)
 SET(TOPSRCDIR "${CMAKE_SOURCE_DIR}")
 SET(TOPBUILDDIR "${CMAKE_BINARY_DIR}")
+SET(VS_CONFIGURATION "/${CMAKE_BUILD_TYPE}")
 configure_file(${CMAKE_SOURCE_DIR}/test_common.in ${CMAKE_BINARY_DIR}/test_common.sh @ONLY NEWLINE_STYLE LF)
 
 #####
diff --git a/configure.ac b/configure.ac
index b0552d60..edef2db1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1333,6 +1333,7 @@ AC_SUBST(HAS_MMAP,[$enable_mmap])
 AC_SUBST(HAS_JNA,[$enable_jna])
 AC_SUBST(RELAX_COORD_BOUND,[$enable_relax_coord_bound])
 AC_SUBST(HAS_ERANGE_FILL,[$enable_erange_fill])
+AC_SUBST(VS_CONFIGURATION,[])
 
 # Include some specifics for netcdf on windows.
 #AH_VERBATIM([_WIN32_STRICMP],
diff --git a/dap4_test/test_data.sh b/dap4_test/test_data.sh
index 07763b26..6549b390 100755
--- a/dap4_test/test_data.sh
+++ b/dap4_test/test_data.sh
@@ -15,8 +15,8 @@ mkdir ./results_test_data
 if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4d ; fi
 for f in $F ; do
     echo "testing: ${f}"
-    if ! ${VG} ${execdir}/test_data ${DAPTESTFILES}/${f} ./results_test_data/${f}.nc ; then
-        failure "${execdir}/test_data ${DAPTESTFILES}/${f} ./results_test_data/${f}.nc"
+    if ! ${VG} ${execdir}${VS}/test_data ${DAPTESTFILES}/${f} ./results_test_data/${f}.nc ; then
+        failure "${execdir}${VS}/test_data ${DAPTESTFILES}/${f} ./results_test_data/${f}.nc"
     fi
     ${NCDUMP} ./results_test_data/${f}.nc > ./results_test_data/${f}.d4d
     if test "x${TEST}" = x1 ; then
diff --git a/dap4_test/test_meta.sh b/dap4_test/test_meta.sh
index 7ab6a8e4..aaa66eb6 100755
--- a/dap4_test/test_meta.sh
+++ b/dap4_test/test_meta.sh
@@ -26,8 +26,8 @@ if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4m ; fi
 mkdir ./results_test_meta
 for f in ${F} ; do
     echo "checking: $f"
-    if ! ${VG} ${execdir}/test_meta ${DMRTESTFILES}/${f}.dmr ./results_test_meta/${f} ; then
-        failure "${execdir}/test_meta ${DMRTESTFILES}/${f}.dmr ./results_test_meta/${f}"
+    if ! ${VG} ${execdir}${VS}/test_meta ${DMRTESTFILES}/${f}.dmr ./results_test_meta/${f} ; then
+        failure "${execdir}${VS}/test_meta ${DMRTESTFILES}/${f}.dmr ./results_test_meta/${f}"
     fi
     ${NCDUMP} -h ./results_test_meta/${f} > ./results_test_meta/${f}.d4m
     if test "x${TEST}" = x1 ; then
diff --git a/dap4_test/test_parse.sh b/dap4_test/test_parse.sh
index a4680fa7..d1e6a4fe 100755
--- a/dap4_test/test_parse.sh
+++ b/dap4_test/test_parse.sh
@@ -15,7 +15,7 @@ mkdir ./results_test_parse
 if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4p ; fi
 for f in $F ; do
     echo "testing: $f"
-    if ! ${VG} ${execdir}/test_parse ${DMRTESTFILES}/${f}.dmr > ./results_test_parse/${f}.d4p ; then
+    if ! ${VG} ${execdir}${VS}/test_parse ${DMRTESTFILES}/${f}.dmr > ./results_test_parse/${f}.d4p ; then
 	failure "${f}"
     fi
     if test "x${TEST}" = x1 ; then
diff --git a/dap4_test/test_remote.sh b/dap4_test/test_remote.sh
index d4ea148c..fc44d0fc 100755
--- a/dap4_test/test_remote.sh
+++ b/dap4_test/test_remote.sh
@@ -54,7 +54,7 @@ test_sequence_2.syn
 test_struct_array.syn
 "
 
-TESTSERVER=`${execdir}/findtestserver4 dap4 d4ts`
+TESTSERVER=`${execdir}${VS}/findtestserver4 dap4 d4ts`
 if test "x$TESTSERVER" = x ; then
 echo "***XFAIL: Cannot find d4ts testserver"
 exit 0
diff --git a/dap4_test/tst_data.sh b/dap4_test/tst_data.sh
index 043b593d..ed84ad82 100755
--- a/dap4_test/tst_data.sh
+++ b/dap4_test/tst_data.sh
@@ -12,8 +12,8 @@ cd $WD
 if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4d ; fi
 for f in $F ; do
     echo "testing: ${f}"
-    if ! ${VG} ${execdir}/test_data ${DAPTESTFILES}/${f} ./results/${f}.nc ; then
-        failure "${execdir}/test_data ${DAPTESTFILES}/${f} ./results/${f}.nc"
+    if ! ${VG} ${execdir}${VS}/test_data ${DAPTESTFILES}/${f} ./results/${f}.nc ; then
+        failure "${execdir}${VS}/test_data ${DAPTESTFILES}/${f} ./results/${f}.nc"
     fi
     ${NCDUMP} ./results/${f}.nc > ./results/${f}.d4d
     if test "x${TEST}" = x1 ; then
diff --git a/dap4_test/tst_meta.sh b/dap4_test/tst_meta.sh
index 7414bee4..73978987 100755
--- a/dap4_test/tst_meta.sh
+++ b/dap4_test/tst_meta.sh
@@ -23,8 +23,8 @@ if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4m ; fi
 
 for f in ${F} ; do
     echo "checking: $f"
-    if ! ${VG} ${execdir}/test_meta ${DMRTESTFILES}/${f}.dmr ./results/${f} ; then
-        failure "${execdir}/test_meta ${DMRTESTFILES}/${f}.dmr ./results/${f}"
+    if ! ${VG} ${execdir}${VS}/test_meta ${DMRTESTFILES}/${f}.dmr ./results/${f} ; then
+        failure "${execdir}${VS}/test_meta ${DMRTESTFILES}/${f}.dmr ./results/${f}"
     fi
     ${NCDUMP} -h ./results/${f} > ./results/${f}.d4m
     if test "x${TEST}" = x1 ; then
diff --git a/dap4_test/tst_parse.sh b/dap4_test/tst_parse.sh
index 99d30ddd..ef8011b7 100755
--- a/dap4_test/tst_parse.sh
+++ b/dap4_test/tst_parse.sh
@@ -12,7 +12,7 @@ cd $WD
 if test "x${RESET}" = x1 ; then rm -fr ${BASELINE}/*.d4p ; fi
 for f in $F ; do
     echo "testing: $f"
-    if ! ${VG} ${execdir}/test_parse ${DMRTESTFILES}/${f}.dmr > ./results/${f}.d4p ; then
+    if ! ${VG} ${execdir}${VS}/test_parse ${DMRTESTFILES}/${f}.dmr > ./results/${f}.d4p ; then
 	failure "${f}"
     fi
     if test "x${TEST}" = x1 ; then
diff --git a/examples/C/run_examples.sh b/examples/C/run_examples.sh
index 655314bf..02b9b9e8 100755
--- a/examples/C/run_examples.sh
+++ b/examples/C/run_examples.sh
@@ -9,16 +9,16 @@ echo "*** Running examples."
 set -e
 
 echo "*** running simple_xy examples..."
-${execdir}/simple_xy_wr
-${execdir}/simple_xy_rd
+${execdir}${VS}/simple_xy_wr
+${execdir}${VS}/simple_xy_rd
 
 echo "*** running sfc_pres_temp examples..."
-${execdir}/sfc_pres_temp_wr
-${execdir}/sfc_pres_temp_rd
+${execdir}${VS}/sfc_pres_temp_wr
+${execdir}${VS}/sfc_pres_temp_rd
 
 echo "*** running pres_temp_4D examples..."
-${execdir}/pres_temp_4D_wr
-${execdir}/pres_temp_4D_rd
+${execdir}${VS}/pres_temp_4D_wr
+${execdir}${VS}/pres_temp_4D_rd
 
 echo "*** Examples successful!"
 exit 0
diff --git a/examples/C/run_examples4.sh b/examples/C/run_examples4.sh
index f0fae095..6ef5ef19 100755
--- a/examples/C/run_examples4.sh
+++ b/examples/C/run_examples4.sh
@@ -9,12 +9,12 @@ echo "*** Running examples for netCDF-4."
 set -e
 
 echo "*** running simple_nc4 examples..."
-${execdir}/simple_nc4_wr
-${execdir}/simple_nc4_rd
+${execdir}${VS}/simple_nc4_wr
+${execdir}${VS}/simple_nc4_rd
 
 echo "*** running simple_xy_nc4 examples..."
-${execdir}/simple_xy_nc4_wr
-${execdir}/simple_xy_nc4_rd
+${execdir}${VS}/simple_xy_nc4_wr
+${execdir}${VS}/simple_xy_nc4_rd
 
 if test -f ${builddir}/findplugin.sh ; then
 echo "*** running test_filter example..."
@@ -30,7 +30,7 @@ export HDF5_PLUGIN_PATH
 
 echo "*** running filter_example..."
 rm -f ./bzip2.nc
-${execdir}/filter_example
+${execdir}${VS}/filter_example
 #rm -f ./bzip2.nc
 
 fi # Filter enabled
diff --git a/examples/C/run_filter.sh b/examples/C/run_filter.sh
index 1a496c16..6c3c4588 100755
--- a/examples/C/run_filter.sh
+++ b/examples/C/run_filter.sh
@@ -22,7 +22,7 @@ HDF5_PLUGIN_PATH="${HDF5_PLUGIN_PATH}/.libs"
 fi
 export HDF5_PLUGIN_PATH
 rm -f ./bzip2.nc
-${execdir}/test_filter
+${execdir}${VS}/test_filter
 #rm -f ./bzip2.nc
 
 echo "*** Example successful!"
diff --git a/examples/CDL/do_comps.sh b/examples/CDL/do_comps.sh
index 88f80f91..e25ffe2c 100755
--- a/examples/CDL/do_comps.sh
+++ b/examples/CDL/do_comps.sh
@@ -2,6 +2,8 @@
 # This shell script runs the cmp test on the example programs.
 # $Id: do_comps.sh,v 1.1 2006/06/27 17:44:54 ed Exp $
 
+. ../../test_common.sh
+
 set -e
 
 ##
@@ -11,13 +13,13 @@ set -e
 echo ""
 echo "*** Creating example data files from CDL scripts."
 echo "*** creating simple_xy.nc..."
-../../ncgen/ncgen -b -o simple_xy.nc $srcdir/simple_xy.cdl
+../../ncgen${VS}/ncgen -b -o simple_xy.nc $srcdir/simple_xy.cdl
 
 echo "*** checking sfc_pres_temp.nc..."
-../../ncgen/ncgen -b -o sfc_pres_temp.nc $srcdir/sfc_pres_temp.cdl
+../../ncgen${VS}/ncgen -b -o sfc_pres_temp.nc $srcdir/sfc_pres_temp.cdl
 
 echo "*** checking pres_temp_4D.nc..."
-../../ncgen/ncgen -b -o pres_temp_4D.nc $srcdir/pres_temp_4D.cdl
+../../ncgen${VS}/ncgen -b -o pres_temp_4D.nc $srcdir/pres_temp_4D.cdl
 
 echo "*** All example creations worked!"
 
diff --git a/hdf4_test/run_formatx_hdf4.sh b/hdf4_test/run_formatx_hdf4.sh
index 8fc19928..1016cb09 100755
--- a/hdf4_test/run_formatx_hdf4.sh
+++ b/hdf4_test/run_formatx_hdf4.sh
@@ -17,7 +17,7 @@ echo "*** Testing extended file format output."
 set -e
 
 echo "Creating HDF4 file"
-${execdir}/tst_interops2
+${execdir}${VS}/tst_interops2
 
 echo "Test extended format output for a HDF4 file"
 rm -f tmp_tst_formatx_hdf4
diff --git a/hdf4_test/run_get_hdf4_files.sh b/hdf4_test/run_get_hdf4_files.sh
index 566dee94..815b0260 100755
--- a/hdf4_test/run_get_hdf4_files.sh
+++ b/hdf4_test/run_get_hdf4_files.sh
@@ -29,7 +29,7 @@ done
 
 echo ""
 echo "Running test program to check HDF4 sample files..."
-${execdir}/tst_interops3
+${execdir}${VS}/tst_interops3
 
 echo "SUCCESS!!!"
 
diff --git a/nc_test/run_diskless.sh b/nc_test/run_diskless.sh
index 3f66e077..3bf897dd 100755
--- a/nc_test/run_diskless.sh
+++ b/nc_test/run_diskless.sh
@@ -21,20 +21,20 @@ echo "*** Testing in-memory (diskless) files with and without persistence"
 HASNC4=`${top_builddir}/nc-config --has-nc4`
 
 echo "**** Test diskless netCDF classic file without persistence"
-${execdir}/tst_diskless
+${execdir}${VS}/tst_diskless
 echo "PASS: diskless netCDF classic file without persistence"
 
 if test "x$HASNC4" = "xyes" ; then
 echo ""
 echo "**** Test diskless netCDF enhanced file without persistence"
-${execdir}/tst_diskless netcdf4
+${execdir}${VS}/tst_diskless netcdf4
 echo "PASS: diskless netCDF enhanced file without persistence"
 fi #HASNC4
 
 echo ""
 echo "**** Test diskless netCDF classic file with persistence"
 rm -f $FILE1
-${execdir}/tst_diskless persist
+${execdir}${VS}/tst_diskless persist
 if test -f $FILE1 ; then
 echo "**** $FILE1 created"
 # ${NCDUMP} $FILE1
@@ -48,7 +48,7 @@ if test "x$HASNC4" = "xyes" ; then
 echo ""
 echo "**** Test diskless netCDF enhanced file with persistence"
 rm -f $FILE1
-${execdir}/tst_diskless netcdf4 persist
+${execdir}${VS}/tst_diskless netcdf4 persist
 if test -f $FILE1 ; then
 echo "**** $FILE1 created"
 # ${NCDUMP} $FILE1
@@ -67,7 +67,7 @@ ok=""
 echo ""
 echo "**** Test extended enhanced diskless netCDF with persistence"
 rm -f $FILE2 tst_diskless2.cdl
-${execdir}/tst_diskless2
+${execdir}${VS}/tst_diskless2
 if test -f $FILE2 ; then
   echo "**** $FILE2 created"
   # Do a cyle test
@@ -102,13 +102,13 @@ rm -f tst_diskless3_file.cdl tst_diskless3_memory.cdl
 echo ""
 echo "**** Create and modify file without using diskless"
 rm -f $FILE3
-${execdir}/tst_diskless3
+${execdir}${VS}/tst_diskless3
 ${NCDUMP} $FILE3 >tst_diskless3_file.cdl
 
 echo ""
 echo "**** Create and modify file using diskless"
 rm -f $FILE3
-${execdir}/tst_diskless3 diskless
+${execdir}${VS}/tst_diskless3 diskless
 ${NCDUMP} $FILE3 >tst_diskless3_memory.cdl
 
 # compare
diff --git a/nc_test/run_diskless5.sh b/nc_test/run_diskless5.sh
index 858e5407..858bacf8 100755
--- a/nc_test/run_diskless5.sh
+++ b/nc_test/run_diskless5.sh
@@ -15,7 +15,7 @@ rm -f $FILE5
 ${NCGEN} -3 -lb -o ${FILE5} ${srcdir}/${CDL5}
 
 echo ""
-${execdir}/tst_diskless5
+${execdir}${VS}/tst_diskless5
 
 # cleanup
 rm -f $FILE5
diff --git a/nc_test/run_mmap.sh b/nc_test/run_mmap.sh
index c9a6ff93..9d1ee7bd 100755
--- a/nc_test/run_mmap.sh
+++ b/nc_test/run_mmap.sh
@@ -19,13 +19,13 @@ echo ""
 echo "*** Testing in-memory (diskless) files with mmap"
 
 echo "**** Test diskless+mmap netCDF classic file without persistence"
-${execdir}/tst_diskless mmap
+${execdir}${VS}/tst_diskless mmap
 echo "PASS: diskless+mmap netCDF classic file without persistence"
 
 echo ""
 echo "**** Test diskless+mmap netCDF classic file with persistence"
 rm -f $FILE1
-${execdir}/tst_diskless mmap persist
+${execdir}${VS}/tst_diskless mmap persist
 if test -f $FILE1 ; then
 echo "**** $FILE1 created"
 # ${NCDUMP} $FILE1
@@ -46,13 +46,13 @@ rm -f tst_diskless3_file.cdl tst_diskless3_memory.cdl
 echo ""
 echo "**** Create and modify file without using diskless+mmap"
 rm -f $FILE3
-${execdir}/tst_diskless3
+${execdir}${VS}/tst_diskless3
 ${NCDUMP} $FILE3 >tst_diskless3_file.cdl
 
 echo ""
 echo "**** Create and modify file using diskless+mmap"
 rm -f $FILE3
-${execdir}/tst_diskless3 diskless mmap
+${execdir}${VS}/tst_diskless3 diskless mmap
 ${NCDUMP} $FILE3 >tst_diskless3_memory.cdl
 
 # compare
diff --git a/nc_test4/filter_test/tst_filter.sh b/nc_test4/filter_test/tst_filter.sh
index 5f59e59f..621f6664 100644
--- a/nc_test4/filter_test/tst_filter.sh
+++ b/nc_test4/filter_test/tst_filter.sh
@@ -38,7 +38,7 @@ sed -e 's/[ 	]*\([^ 	].*\)/\1/' <$1 >$2
 if test "x$API" = x1 ; then
 echo "*** Testing dynamic filters using API"
 rm -f ./bzip2.nc ./bzip2.dump ./tmp
-${execdir}/test_filter
+${execdir}${VS}/test_filter
 $NCDUMP -s ./bzip2.nc > ./tmp
 # Remove irrelevant -s output
 sclean ./tmp ./bzip2.dump
@@ -50,7 +50,7 @@ if test "x$MISC" = x1 ; then
 echo
 echo "*** Testing dynamic filters parameter passing"
 rm -f ./testmisc.nc tmp tmp2
-${execdir}/test_misc
+${execdir}${VS}/test_misc
 # Verify the parameters via ncdump
 $NCDUMP -s ./testmisc.nc > ./tmp
 # Extract the parameters
diff --git a/nc_test4/run_bm.sh b/nc_test4/run_bm.sh
index 2b8d0905..f7c3a640 100755
--- a/nc_test4/run_bm.sh
+++ b/nc_test4/run_bm.sh
@@ -30,7 +30,7 @@ for ((s=0; s < 2 ; s++))
 do
     for ((d=0; d <= 9 ; d=d+2))
     do
-	cmd="${execdir}/bm_file $h -f 3 -o $TMP/$d1-2d.nc4 -c 0:${d}:${s}:${c0}:${c1}"
+	cmd="${execdir}${VS}/bm_file $h -f 3 -o $TMP/$d1-2d.nc4 -c 0:${d}:${s}:${c0}:${c1}"
 	for ((v=1; v < 12; v++))
 	do
 	    cmd="$cmd,${v}:${d}:${s}:${c0}:${c1}"
@@ -62,7 +62,7 @@ for ((s=0; s < 2 ; s++))
 do
     for ((d=0; d <= 9 ; d++))
     do
-	cmd="${execdir}/bm_file $h -f 3 -o $TMP/$d1.nc4 -c 0:${d}:${s}:${c0}:${c1}:${c2} $TMP/$d1.nc3"
+	cmd="${execdir}${VS}/bm_file $h -f 3 -o $TMP/$d1.nc4 -c 0:${d}:${s}:${c0}:${c1}:${c2} $TMP/$d1.nc3"
 	echo "cmd=$cmd"
 	if ! ($cmd >> $out1); then
 	    exit 1;
@@ -82,7 +82,7 @@ do
     do
 	for c3 in 10 100 200 500
 	do
-	    cmd="${execdir}/bm_file $h -f 3 -o $TMP/$d1.nc4 -c 0:${d}:${s}:${c0}:${c1}:${c2} $TMP/$d1.nc3"
+	    cmd="${execdir}${VS}/bm_file $h -f 3 -o $TMP/$d1.nc4 -c 0:${d}:${s}:${c0}:${c1}:${c2} $TMP/$d1.nc3"
 	    echo "cmd=$cmd"
 	    if ! ($cmd >> $out1); then
 		exit 1;
diff --git a/nc_test4/run_bm_ar4.sh b/nc_test4/run_bm_ar4.sh
index 4e4ebc8c..04097e7e 100755
--- a/nc_test4/run_bm_ar4.sh
+++ b/nc_test4/run_bm_ar4.sh
@@ -55,7 +55,7 @@ do
 		if test -f $file_out; then
 		    echo "found existing $file_out"
 		else
-		    cmd="${execdir}/bm_file $h -f 3 -c 6:${d}:${s}:${c0}:${c1}:${c2} -o ${file_out} ${pr_ar4_sample}"
+		    cmd="${execdir}${VS}/bm_file $h -f 3 -c 6:${d}:${s}:${c0}:${c1}:${c2} -o ${file_out} ${pr_ar4_sample}"
 		    echo "cmd=$cmd"
 # 	            bash ./clear_cache.sh
  		    if ! ($cmd >> $out); then
@@ -74,20 +74,20 @@ echo $out_files
 
 # Do the horizonatal runs.
 #bash ./clear_cache.sh
-${execdir}/tst_ar4 -h $pr_ar4_sample
+${execdir}${VS}/tst_ar4 -h $pr_ar4_sample
 for f1 in $out_files
 do
 #    bash ./clear_cache.sh
-    ${execdir}/tst_ar4 ${f1}
+    ${execdir}${VS}/tst_ar4 ${f1}
 done
 
 # Do the timeseries runs.
 #bash ./clear_cache.sh
-${execdir}/tst_ar4 -t -h $pr_ar4_sample
+${execdir}${VS}/tst_ar4 -t -h $pr_ar4_sample
 for f1 in $out_files
 do
 #    bash ./clear_cache.sh
-    ${execdir}/tst_ar4 -t ${f1}
+    ${execdir}${VS}/tst_ar4 -t ${f1}
 done
 
 echo "SUCCESS!!!"
diff --git a/nc_test4/run_bm_elena.sh b/nc_test4/run_bm_elena.sh
index cf8c1756..2902c5dc 100755
--- a/nc_test4/run_bm_elena.sh
+++ b/nc_test4/run_bm_elena.sh
@@ -9,12 +9,12 @@ set -e
 echo ""
 
 echo "*** Testing the benchmarking program bm_file for simple float file, no compression..."
-${execdir}/bm_file -h -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:1024:16:256 tst_elena_int_3D.nc
-${execdir}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:1024:256:256 tst_elena_int_3D.nc
-${execdir}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:512:64:256 tst_elena_int_3D.nc
-${execdir}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:512:256:256 tst_elena_int_3D.nc
-${execdir}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:256:64:256 tst_elena_int_3D.nc
-${execdir}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:256:256:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -h -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:1024:16:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:1024:256:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:512:64:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:512:256:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:256:64:256 tst_elena_int_3D.nc
+${execdir}${VS}/bm_file -d -f 3 -o  tst_elena_out.nc -c 0:-1:0:256:256:256 tst_elena_int_3D.nc
 echo '*** SUCCESS!!!'
 
 exit 0
diff --git a/nc_test4/run_bm_test1.sh b/nc_test4/run_bm_test1.sh
index 0a6ed37e..0d117393 100755
--- a/nc_test4/run_bm_test1.sh
+++ b/nc_test4/run_bm_test1.sh
@@ -24,7 +24,7 @@ do
 	test $i = 4 && chunksizes="17:17:17:17" && s="-s 1"
 	test $i = 5 && chunksizes="10:10:10:10:10" && s="-s 1"
 	test $i = 6 && chunksizes="6:6:6:6:6:6" && s="-s 1"
-	${execdir}/bm_file -d ${header} ${s} -f 4 -o ${type_name}2_${i}D_3.nc -c 0:0:0:${chunksizes} tst_${type_name}2_${i}D.nc 
+	${execdir}${VS}/bm_file -d ${header} ${s} -f 4 -o ${type_name}2_${i}D_3.nc -c 0:0:0:${chunksizes} tst_${type_name}2_${i}D.nc 
 	${NCDUMP} tst_${type_name}2_${i}D.nc > tst_${type_name}2_${i}D.cdl
 	${NCDUMP} -n tst_${type_name}2_${i}D ${type_name}2_${i}D_3.nc > ${type_name}2_${i}D.cdl
 	diff tst_${type_name}2_${i}D.cdl ${type_name}2_${i}D.cdl
@@ -35,7 +35,7 @@ done
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, no compression..."
-${execdir}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:-1:0:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:-1:0:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
@@ -43,7 +43,7 @@ echo '*** SUCCESS!!!'
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, with compression..."
-${execdir}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:1:0:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:1:0:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
@@ -51,7 +51,7 @@ echo '*** SUCCESS!!!'
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, with more compression..."
-${execdir}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
@@ -59,7 +59,7 @@ echo '*** SUCCESS!!!'
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, with endianness set to native..."
-${execdir}/bm_file -e 0 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -e 0 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
@@ -67,7 +67,7 @@ echo '*** SUCCESS!!!'
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, with endianness set to little..."
-${execdir}/bm_file -e 1 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -e 1 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
@@ -75,7 +75,7 @@ echo '*** SUCCESS!!!'
 
 echo ""
 echo "*** Testing the benchmarking program bm_file for simple float file, with endianness set to big..."
-${execdir}/bm_file -e 2 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
+${execdir}${VS}/bm_file -e 2 -d -h -f 3 -o floats_1D_3.nc -c 0:9:1:10000 tst_floats_1D.nc
 ${NCDUMP} tst_floats_1D.nc > tst_floats_1D.cdl
 ${NCDUMP} -n tst_floats_1D floats_1D_3.nc > floats_1D.cdl
 diff tst_floats_1D.cdl floats_1D.cdl
diff --git a/nc_test4/run_bm_test2.sh b/nc_test4/run_bm_test2.sh
index ed747b1c..c5ed714d 100755
--- a/nc_test4/run_bm_test2.sh
+++ b/nc_test4/run_bm_test2.sh
@@ -11,28 +11,28 @@ set -e
 echo ""
 
 echo "*** Testing the benchmarking program bm_file using start/count/inc..."
-${execdir}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:1 -r 0:1 tst_simple.nc
+${execdir}${VS}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:1 -r 0:1 tst_simple.nc
 ${NCDUMP} tst_simple.nc > tst_simple.cdl
 ${NCDUMP} -n tst_simple tst_simple_3.nc > tst_simple_3.cdl
 diff tst_simple.cdl tst_simple_3.cdl
 echo '*** SUCCESS!!!'
 
 echo "*** Testing the benchmarking program bm_file using start/count/inc with negative increment..."
-${execdir}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:9 -u 0:1 -r 0:-1 tst_simple.nc
+${execdir}${VS}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:9 -u 0:1 -r 0:-1 tst_simple.nc
 ${NCDUMP} tst_simple.nc > tst_simple.cdl
 ${NCDUMP} -n tst_simple tst_simple_3.nc > tst_simple_3.cdl
 diff tst_simple.cdl tst_simple_3.cdl
 echo '*** SUCCESS!!!'
 
 echo "*** Testing the benchmarking program bm_file using start/count/inc with count > 1..."
-${execdir}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:2 -r 0:2 tst_simple.nc
+${execdir}${VS}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:2 -r 0:2 tst_simple.nc
 ${NCDUMP} tst_simple.nc > tst_simple.cdl
 ${NCDUMP} -n tst_simple tst_simple_3.nc > tst_simple_3.cdl
 diff tst_simple.cdl tst_simple_3.cdl
 echo '*** SUCCESS!!!'
 
 echo "*** Testing the benchmarking program bm_file using start/count/inc with uneven count..."
-${execdir}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:4 -r 0:4 tst_simple.nc
+${execdir}${VS}/bm_file -h -f 3 -o tst_simple_3.nc -c 0:-1:0:10  -t 0:0 -u 0:4 -r 0:4 tst_simple.nc
 ${NCDUMP} tst_simple.nc > tst_simple.cdl
 ${NCDUMP} -n tst_simple tst_simple_3.nc > tst_simple_3.cdl
 diff tst_simple.cdl tst_simple_3.cdl
diff --git a/nc_test4/run_empty_vlen_test.sh b/nc_test4/run_empty_vlen_test.sh
index 9cb057b5..e9a65bde 100755
--- a/nc_test4/run_empty_vlen_test.sh
+++ b/nc_test4/run_empty_vlen_test.sh
@@ -20,7 +20,7 @@ echo ""
 echo "* Testing Empty Ragged Arrays (VLEN)"
 
 echo "Generating test netcdf files."
-${execdir}/tst_empty_vlen_unlim
+${execdir}${VS}/tst_empty_vlen_unlim
 
 # Since no comparison is made, I am not sure 
 # if this is useful.
diff --git a/nc_test4/run_grp_rename.sh b/nc_test4/run_grp_rename.sh
index ab5a72ca..a3f52e41 100755
--- a/nc_test4/run_grp_rename.sh
+++ b/nc_test4/run_grp_rename.sh
@@ -86,13 +86,13 @@ rm -f tst_grp_rename.nc
 ${NCGEN} -k nc4 ./tst_grp_rename.cdl
 
 # Try to rename 2nd level group
-if ! ${execdir}/renamegroup tst_grp_rename.nc "inner/inner_inner" "inner_renamed" ; then
+if ! ${execdir}${VS}/renamegroup tst_grp_rename.nc "inner/inner_inner" "inner_renamed" ; then
   echo "***FAIL: attempt to rename /inner/inner_inner failed"
   FAIL=1
 fi
 
 # Try to 1st level group
-if ! ${execdir}/renamegroup tst_grp_rename.nc "inner" "renamed" ; then
+if ! ${execdir}${VS}/renamegroup tst_grp_rename.nc "inner" "renamed" ; then
   echo "***FAIL: attempt to rename /inner failed"
   FAIL=1
 fi
@@ -107,7 +107,7 @@ if ! diff -b ref_grp_rename.cdl tst_grp_rename.dmp ; then
 fi
 
 # Finally, try to rename root group; should fail
-if ${execdir}/renamegroup tst_grp_rename.nc "/" "rootgroup" ; then
+if ${execdir}${VS}/renamegroup tst_grp_rename.nc "/" "rootgroup" ; then
   echo "***FAIL: attempt to rename root group should not have succeeded"
   FAIL=1
 else
diff --git a/nc_test4/run_tst_chunks.sh b/nc_test4/run_tst_chunks.sh
index a002471f..dda206e0 100755
--- a/nc_test4/run_tst_chunks.sh
+++ b/nc_test4/run_tst_chunks.sh
@@ -20,7 +20,7 @@ dim2=12
 chunk2=3
 dim3=4
 chunk3=1
-${execdir}/tst_chunks3 $compress_level $dim1 $chunk1 $dim2 $chunk2 $dim3 $chunk3
+${execdir}${VS}/tst_chunks3 $compress_level $dim1 $chunk1 $dim2 $chunk2 $dim3 $chunk3
 ${NCDUMP} -n tst_chunks -s tst_chunks3.nc > tst_chunks3.cdl
 diff tst_chunks3.cdl ref_chunks1.cdl
 echo '*** SUCCESS!!!'
@@ -37,7 +37,7 @@ chunk3=13
 cachesize=10000000
 cachehash=10000
 cachepre=0.0
-${execdir}/tst_chunks3 $compress_level $dim1 $chunk1 $dim2 $chunk2 $dim3 $chunk3 $cachesize $cachehash $cachepre
+${execdir}${VS}/tst_chunks3 $compress_level $dim1 $chunk1 $dim2 $chunk2 $dim3 $chunk3 $cachesize $cachehash $cachepre
 ${NCDUMP} -n tst_chunks -s -h tst_chunks3.nc > tst_chunks3.cdl
 diff tst_chunks3.cdl ref_chunks2.cdl
 echo '*** SUCCESS!!!'
diff --git a/nc_test4/tst_filter.sh b/nc_test4/tst_filter.sh
index 1dc5df1e..df75b990 100755
--- a/nc_test4/tst_filter.sh
+++ b/nc_test4/tst_filter.sh
@@ -56,7 +56,7 @@ if ! test -f ${MISCPATH} ; then echo "Unable to locate ${MISCPATH}"; exit 1; fi
 if test "x$API" = x1 ; then
 echo "*** Testing dynamic filters using API"
 rm -f ./bzip2.nc ./bzip2.dump ./tst_filter.txt
-${execdir}/test_filter
+${execdir}${VS}/test_filter
 ${NCDUMP} -s bzip2.nc > ./tst_filter.txt
 # Remove irrelevant -s output
 sclean ./tst_filter.txt ./bzip2.dump
@@ -68,7 +68,7 @@ if test "x$MISC" = x1 ; then
 echo
 echo "*** Testing dynamic filters parameter passing"
 rm -f ./testmisc.nc tst_filter.txt tst_filter2.txt
-${execdir}/test_filter_misc
+${execdir}${VS}/test_filter_misc
 # Verify the parameters via ncdump
 ${NCDUMP} -s testmisc.nc > ./tst_filter.txt
 # Extract the parameters
diff --git a/nc_test4/tst_szip.sh b/nc_test4/tst_szip.sh
index dfffdb0c..ed906db4 100755
--- a/nc_test4/tst_szip.sh
+++ b/nc_test4/tst_szip.sh
@@ -12,15 +12,15 @@ ${NCDUMP} ${srcdir}/ref_szip.h5 >szip_dump.cdl
 diff -w ${srcdir}/ref_szip.cdl ./szip_dump.cdl
 
 echo "*** Testing tst_szip "
-${execdir}/test_szip
+${execdir}${VS}/test_szip
 echo "***Passed"
 
 echo "*** Testing h5testszip "
-${execdir}/h5testszip
+${execdir}${VS}/h5testszip
 echo "***Passed"
 
 echo "*** Testing h5testszip on testszip.nc"
-${execdir}/h5testszip ./testszip.nc
+${execdir}${VS}/h5testszip ./testszip.nc
 echo "***Passed"
 
 rm -f testnc.h5 testszip.nc
diff --git a/ncdap_test/tst_remote.sh b/ncdap_test/tst_remote.sh
index 64789b1f..cfa688ac 100755
--- a/ncdap_test/tst_remote.sh
+++ b/ncdap_test/tst_remote.sh
@@ -8,7 +8,7 @@ leakcheck=0
 timing=0
 
 # Figure our dst server; if none, then just stop
-DTS=`${execdir}/findtestserver dap2 dts`
+DTS=`${execdir}${VS}/findtestserver dap2 dts`
 if test "x$DTS" = "x" ; then
 echo "WARNING: Cannot locate test server for dts"
 exit
diff --git a/ncdump/run_utf8_tests.sh b/ncdump/run_utf8_tests.sh
index 679f8b13..d50ee92a 100755
--- a/ncdump/run_utf8_tests.sh
+++ b/ncdump/run_utf8_tests.sh
@@ -10,7 +10,7 @@ echo ""
 echo "*** Testing ncgen and ncdump for UTF8 support..."
 
 # Run tst_utf8.c to produce test file tst_utf8.nc.
-${execdir}/tst_utf8
+${execdir}${VS}/tst_utf8
 
 # if test "x$builddir" = x ; then
 # builddir="."
diff --git a/ncdump/tst_bom.sh b/ncdump/tst_bom.sh
index 40232a38..f533c3f6 100755
--- a/ncdump/tst_bom.sh
+++ b/ncdump/tst_bom.sh
@@ -25,7 +25,7 @@ data:
 EOF
 
 echo "*** Generate a cdl file with leading UTF-8 BOM."
-${execdir}/bom 8 >tst_bom8.cdl
+${execdir}${VS}/bom 8 >tst_bom8.cdl
 cat tst_bom.cdl >> tst_bom8.cdl
 
 echo "*** Verify .nc file"
@@ -38,7 +38,7 @@ diff -w tst_bom.cdl tmp.cdl
 rm -f tmp.cdl tst_bom8.* tst_bom16.*
 
 echo "*** Generate a cdl file with leading UTF-16 BOM."
-${execdir}/bom 16 >tst_bom16.cdl
+${execdir}${VS}/bom 16 >tst_bom16.cdl
 cat tst_bom.cdl >> tst_bom16.cdl
 
 echo "*** Verify UTF-16 file fails"
diff --git a/ncdump/tst_dimsizes.sh b/ncdump/tst_dimsizes.sh
index e08faba8..57546a3e 100755
--- a/ncdump/tst_dimsizes.sh
+++ b/ncdump/tst_dimsizes.sh
@@ -16,7 +16,7 @@ echo ""
 rm -f tst_dimsize_classic.nc tst_dimsize_64offset.nc tst_dimsize_64data.nc
 
 echo "*** Generate: tst_dimsize_classic.nc tst_dimsize_64offset.nc tst_dimsize_64data.nc"
-${execdir}/tst_dimsizes
+${execdir}${VS}/tst_dimsizes
 
 echo "*** Verify that ncdump can read dimsizes"
 
diff --git a/ncdump/tst_fileinfo.sh b/ncdump/tst_fileinfo.sh
index 87e0ee0c..dd4dc484 100755
--- a/ncdump/tst_fileinfo.sh
+++ b/ncdump/tst_fileinfo.sh
@@ -21,7 +21,7 @@ else
 fi
 rm -f ./tmp_tst_fileinfo
 
-if ${execdir}/tst_fileinfo > /dev/null ; then
+if ${execdir}${VS}/tst_fileinfo > /dev/null ; then
    # look at the _IsNetcdf4 flag
    N_IS=`${NCDUMP} -s $NCF | fgrep '_IsNetcdf4' | tr -d ' ;'`
    N_IS=`echo $N_IS | cut -d= -f2`
diff --git a/ncdump/tst_fillbug.sh b/ncdump/tst_fillbug.sh
index f83081f9..71cee22d 100755
--- a/ncdump/tst_fillbug.sh
+++ b/ncdump/tst_fillbug.sh
@@ -8,7 +8,7 @@ if test "x$srcdir" = x ; then srcdir=`pwd`; fi
 echo ""
 echo "*** Running ncdump bug test."
 
-${execdir}/tst_fillbug
+${execdir}${VS}/tst_fillbug
 # echo "*** dumping tst_fillbug.nc to tst_fillbug.cdl..."
 ${NCDUMP} tst_fillbug.nc > tst_fillbug.cdl
 # echo "*** comparing tst_fillbug.cdl with ref_tst_fillbug.cdl..."
diff --git a/ncdump/tst_h_scalar.sh b/ncdump/tst_h_scalar.sh
index f5b6698e..efb111eb 100755
--- a/ncdump/tst_h_scalar.sh
+++ b/ncdump/tst_h_scalar.sh
@@ -9,7 +9,7 @@ set -e
 echo ""
 echo "*** Running ncdump scalar test."
 
-${execdir}/tst_h_scalar
+${execdir}${VS}/tst_h_scalar
 # echo "*** dumping tst_h_scalar.nc to tst_h_scalar.cdl..."
 ${NCDUMP} tst_h_scalar.nc > tst_h_scalar.cdl
 # echo "*** comparing tst_h_scalar.cdl with ref_tst_h_scalar.cdl..."
diff --git a/ncdump/tst_lengths.sh b/ncdump/tst_lengths.sh
index dbe5c681..104422ac 100755
--- a/ncdump/tst_lengths.sh
+++ b/ncdump/tst_lengths.sh
@@ -70,13 +70,13 @@ fi
 #fi
 
 echo "*** testing length of rewritten classic file"
-${NCGEN} -b ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+${NCGEN} -b ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
 #if test `wc -c < small.nc` != 68; then
 #    exit 1
 #fi
 
 echo "*** testing length of rewritten classic file written with NOFILL"
-${NCGEN} -b -x ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+${NCGEN} -b -x ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
 #if test `wc -c < small.nc` != 68; then
 #    exit 1
 #fi
@@ -94,13 +94,13 @@ ${NCGEN} -b -k64-bit-offset -x ${srcdir}/small.cdl
 #fi
 
 echo "*** testing length of rewritten 64-bit offset file"
-${NCGEN} -b -k64-bit-offset ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+${NCGEN} -b -k64-bit-offset ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
 #if test `wc -c < small.nc` != 72; then
 #    exit 1
 #fi
 
 echo "*** testing length of rewritten 64-bit offset file written with NOFILL"
-${NCGEN} -b -k64-bit-offset -x ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+${NCGEN} -b -k64-bit-offset -x ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
 #if test `wc -c < small.nc` != 72; then
 #    exit 1
 #fi
@@ -126,16 +126,16 @@ if test "x$HAVE_CDF5" = x1 ; then
     #fi
 
     echo "*** testing length of rewritten 64-bit data file"
-    ${NCGEN} -b -5 ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+    ${NCGEN} -b -5 ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
     # Watch out, it appears that the CDF-5 files are being rounded up to next page size
     # So, we need to truncate them wrt nul's in order to check size.
     # Bad hack, but what else can I do?
-    if test `${execdir}/nctrunc <small.nc |wc -c` != 104; then
+    if test `${execdir}${VS}/nctrunc <small.nc |wc -c` != 104; then
         exit 1
     fi
 
     echo "*** testing length of rewritten 64-bit data file written with NOFILL"
-    ${NCGEN} -b -5 -x ${srcdir}/small.cdl && ${execdir}/rewrite-scalar small.nc t
+    ${NCGEN} -b -5 -x ${srcdir}/small.cdl && ${execdir}${VS}/rewrite-scalar small.nc t
     #if test `wc -c < small.nc` != 104; then
     #    exit 1
     #fi
diff --git a/ncdump/tst_nccopy4.sh b/ncdump/tst_nccopy4.sh
index c4668a43..350421bc 100755
--- a/ncdump/tst_nccopy4.sh
+++ b/ncdump/tst_nccopy4.sh
@@ -17,8 +17,8 @@ TESTFILES='tst_comp tst_comp2 tst_enum_data tst_fillbug
  tst_vlen_data'
 
 # Run these programs to create some test files.
-${execdir}/tst_comp2
-${execdir}/tst_compress
+${execdir}${VS}/tst_comp2
+${execdir}${VS}/tst_compress
 
 echo "*** Testing netCDF-4 features of nccopy on ncdump/*.nc files"
 for i in $TESTFILES ; do
diff --git a/ncdump/tst_netcdf4.sh b/ncdump/tst_netcdf4.sh
index 4c126025..9cf74b5c 100755
--- a/ncdump/tst_netcdf4.sh
+++ b/ncdump/tst_netcdf4.sh
@@ -25,10 +25,10 @@ echo "*** Creating test output tst_netcdf4_c0.nc."
 ${NCGEN} -k nc7 -b -o tst_netcdf4_c0.nc ${ncgenc0} ; ERR
 
 echo "*** Testing that program tst_h_rdc0 can read tst_netcdf4_c0.nc."
-${execdir}/tst_h_rdc0 ; ERR
+${execdir}${VS}/tst_h_rdc0 ; ERR
 
 echo "*** Running tst_create_files.c to create test files."
-${execdir}/tst_create_files ; ERR
+${execdir}${VS}/tst_create_files ; ERR
 echo "*** Testing tst_create_files output for netCDF-4 features."
 ${NCDUMP} tst_solar_1.nc | sed 's/e+0/e+/g' > tst_solar_1.cdl ; ERR
 diff -b tst_solar_1.cdl $srcdir/ref_tst_solar_1.cdl ; ERR
@@ -36,7 +36,7 @@ ${NCDUMP} tst_solar_2.nc | sed 's/e+0/e+/g' > tst_solar_2.cdl ; ERR
 diff -b tst_solar_2.cdl $srcdir/ref_tst_solar_2.cdl ; ERR
 
 echo "*** Running tst_group_data.c to create test files."
-${execdir}/tst_group_data ; ERR
+${execdir}${VS}/tst_group_data ; ERR
 ${NCDUMP} tst_group_data.nc | sed 's/e+0/e+/g' > tst_group_data.cdl ; ERR
 diff -b tst_group_data.cdl $srcdir/ref_tst_group_data.cdl ; ERR
 
@@ -49,41 +49,41 @@ ${NCDUMP} -v var,var2 tst_group_data.nc | sed 's/e+0/e+/g' > tst_group_data.cdl
 diff -b tst_group_data.cdl $srcdir/ref_tst_group_data.cdl ; ERR
 
 echo "*** Running tst_enum_data.c to create test files."
-${execdir}/tst_enum_data ; ERR
+${execdir}${VS}/tst_enum_data ; ERR
 ${NCDUMP} tst_enum_data.nc | sed 's/e+0/e+/g' > tst_enum_data.cdl ; ERR
 diff -b tst_enum_data.cdl $srcdir/ref_tst_enum_data.cdl ; ERR
 
 echo "*** Running tst_opaque_data.c to create test files."
-${execdir}/tst_opaque_data ; ERR
+${execdir}${VS}/tst_opaque_data ; ERR
 ${NCDUMP} tst_opaque_data.nc | sed 's/e+0/e+/g' > tst_opaque_data.cdl ; ERR
 diff -b tst_opaque_data.cdl $srcdir/ref_tst_opaque_data.cdl ; ERR
 
 echo "*** Running tst_vlen_data.c to create test files."
-${execdir}/tst_vlen_data ; ERR
+${execdir}${VS}/tst_vlen_data ; ERR
 ${NCDUMP} tst_vlen_data.nc | sed 's/e+0/e+/g' > tst_vlen_data.cdl ; ERR
 diff -b tst_vlen_data.cdl $srcdir/ref_tst_vlen_data.cdl ; ERR
 
 echo "*** Running tst_comp.c to create test files."
-${execdir}/tst_comp ; ERR
+${execdir}${VS}/tst_comp ; ERR
 ${NCDUMP} tst_comp.nc | sed 's/e+0/e+/g' > tst_comp.cdl ; ERR
 diff -b tst_comp.cdl $srcdir/ref_tst_comp.cdl ; ERR
 
 echo "*** Running tst_nans.c to create test files."
-${execdir}/tst_nans ; ERR
+${execdir}${VS}/tst_nans ; ERR
 ${NCDUMP} tst_nans.nc | sed 's/e+0/e+/g' > tst_nans.cdl ; ERR
 diff -b tst_nans.cdl $srcdir/ref_tst_nans.cdl ; ERR
 
 # Do unicode test only if it exists => BUILD_UTF8 is true
-if test -f ./tst_unicode -o -f ./tst_unicode.exe ; then
+if test -f ./tst_unicode -o -f ./tst_unicode.exe -o -f ${execdir}${VS}/tst_unicode.exe; then
   echo "*** dumping tst_unicode.nc to tst_unicode.cdl..."
-  ${execdir}/tst_unicode ; ERR
+  ${execdir}${VS}/tst_unicode ; ERR
 ${NCDUMP} tst_unicode.nc | sed 's/e+0/e+/g' > tst_unicode.cdl ; ERR
   #echo "*** comparing tst_unicode.cdl with ref_tst_unicode.cdl..."
   #diff -b tst_unicode.cdl $srcdir/ref_tst_unicode.cdl
 fi
 
 echo "*** Running tst_special_atts.c to create test files."
-${execdir}/tst_special_atts ; ERR
+${execdir}${VS}/tst_special_atts ; ERR
 ${NCDUMP} -c -s tst_special_atts.nc \
     | sed 's/e+0/e+/g' \
     | sed -e 's/netcdflibversion=.*[|]/netcdflibversion=0.0.0|/' \
diff --git a/ncdump/tst_netcdf4_4.sh b/ncdump/tst_netcdf4_4.sh
index 4da9b4b3..17726516 100755
--- a/ncdump/tst_netcdf4_4.sh
+++ b/ncdump/tst_netcdf4_4.sh
@@ -22,7 +22,7 @@ echo "*** Running extra netcdf-4 tests."
 #
 
 echo "*** running tst_string_data to create test files..."
-${execdir}/tst_string_data
+${execdir}${VS}/tst_string_data
 
 if [ `uname | cut -d "_" -f 1` = "MINGW32" ]; then # MINGW Platforms
 
diff --git a/ncdump/tst_output.sh b/ncdump/tst_output.sh
index 87a53186..a4d589a8 100755
--- a/ncdump/tst_output.sh
+++ b/ncdump/tst_output.sh
@@ -10,7 +10,7 @@ echo ""
 echo "*** Testing ncgen and ncdump test output for classic format."
 
 echo "*** Testing that ncgen produces correct C code from c0.cdl."
-${execdir}/ref_ctest
+${execdir}${VS}/ref_ctest
 ${NCGEN} -lc -o ctest0.nc $srcdir/../ncgen/c0.cdl > tst_output_ctest.c
 diff -b tst_output_ctest.c $srcdir/ref_ctest.c
 
@@ -49,7 +49,7 @@ diff -b tst_format_att.cdl $srcdir/ref_tst_format_att.cdl
 echo "*** All ncgen and ncdump test output for classic format passed!"
 
 echo "*** Testing that ncgen with c0.cdl for 64-bit offset format."
-${execdir}/ref_ctest64
+${execdir}${VS}/ref_ctest64
 ${NCGEN}  -k2 -lc -o ctest0_64.nc $srcdir/../ncgen/c0.cdl > tst_output_ctest64.c
 diff -b tst_output_ctest64.c $srcdir/ref_ctest64.c
 
diff --git a/test_common.in b/test_common.in
index b8af5185..45aebb39 100644
--- a/test_common.in
+++ b/test_common.in
@@ -59,8 +59,8 @@ if test "x$SETX" = x1 ; then set -x ; fi
 top_srcdir="$TOPSRCDIR"
 top_builddir="$TOPBUILDDIR"
 
-# Currently not used, but left as a Visual Studio placeholder.
-# VS=Debug
+# Only set via CMake builds, where it must contain a leading '/'
+export VS=@VS_CONFIGURATION@
 
 # srcdir may or may not be defined, but if not, then create it
 if test "x$srcdir" = x ; then
-- 
2.15.0

